addon/mixins/flexberry-maplayer-actions-handler.js

```diff
@@ -384,33 +385,59 @@ export default Ember.Mixin.create({
   },
 
   /**
-   Sets indexes for layers hierarchy.
+    Sets indexes for layers hierarchy.
 
-    @method setIndexes
-    @param {Object} rootLayer Root layer of the tree.
-    @param {Boolean} isAdded True when layer is added.
+    @method _setIndexes
+    @param {Array} layers Array of layers to set indexes.
     @param {Int} index First index.
     @returns {Int} Last index.
+    @private
   */
-  setIndexes(rootLayer, isAdded, index) {
-    let layers = rootLayer.get('layers');
-    if (Ember.isArray(layers)) {
-      if (Ember.isNone(index)) {
-        index = layers.objectAt(0).get('index') || 0;
-        index = isAdded ? index + 1 : index - 1;
-      }
-
+  _setIndexes(layers, index) {
+    if (Ember.isArray(layers) && index > 0) {
       layers.forEach((layer) => {
         if (!layer.get('isDeleted')) {
           layer.set('index', index);
           index--;
-          if (layer.get('type') === 'group') {
-            index = this.setIndexes(layer, isAdded, index);
+          if (Ember.isArray(layer.get('layers'))) {
+            index = this._setIndexes(layer.get('layers'), index);
           }
         }
       }, this);
     }
 
     return index;
   },
+
+  /**
+    Finds root layers.
+
+    @method _findRootLayers
+    @param {String} layerPath LayerPath to child layer.
+    @returns {Array} Root layers.
+    @private
+  */
+  _findRootLayers(layerPath) {
+    let layer = getRecord(this, layerPath);
+    let nextInPath = function(path) {
+      let index = path.lastIndexOf('.');
+      if (index > -1) {
+        path = path.slice(0, index);
+      }
+
+      return path;
+    };
+
+    // Get layerPath for layers on same hierarchy level.
+    layerPath = nextInPath(layerPath);
+
+    if (Ember.isNone(layer.get('parent'))) {
```

Comment:
I suggest it is better to use Ember.isBlank cuz it returns TRUE both for none (undefined, null) and empty objects (empty strings, whitespaced strings, empty arrays).
see: https://emberjs.com/api/#method_isBlank
It is just for safety, cuz we already had such a problem with edit-map controller's queryFilter, as I remember